sw/device/examples/hello_usbdev/hello_usbdev.c

```diff
@@ -33,42 +53,32 @@ static usbdev_ctx_t usbdev_ctx;
 static usb_controlep_ctx_t control_ctx;
 static usb_ss_ctx_t ss_ctx[2];
 
-static void test_error(void) {
-  while (1) {
-    gpio_write_all(0xAA00);  // pattern
-    busy_sleep_micros(200 * 1000);
-    gpio_write_all(0x5500);  // pattern
-    busy_sleep_micros(100 * 1000);
-  }
-}
-
-// Override default handler routines
-void handler_instr_ill_fault(void) {
-  uart_send_str("Instruction Illegal fault");
-  test_error();
-}
-
-// Override default handler routines
-void handler_lsu_fault(void) {
-  uart_send_str("Load/Store Fault");
-  test_error();
-}
-
-#define MK_PRINT(c) \
-  (((c != 0xa) && (c != 0xd) && ((c < 32) || (c > 126))) ? '_' : c)
+// We stop the application after receiving USB_MAX.
+static volatile unsigned int num_chars_rx = 0;
+#ifndef USB_MAX
+#define USB_MAX 6
```

Comment:
This value is very low for reasons of CI checks.

Reply:
That's fine, in our system I stop the simulations at frame=35, which is around the same time. With usbdpi nothing interesting happens after that.

Does your CI pipeline have a short timeout? Otherwise it might make sense to abort the simulation if nothing happened after x ms.

Reply:
The simulation just needs to signal to the pipeline when it was successful or failed. We currently do this by printing "PASS!" or "FAIL!" on the UART. I have now changed the application such that the demo loop continues after signaling "PASS!".